### DISCUSSION TITLE
Ability to use custom parser?

### DISCUSSION THREAD

**Discussion by kdy1** [babel/babel#13228] [Ideas]
Hi. I'm the author of the [swc][] project.

I want to know if there's an option to replace parser completely.
I searched for the babel parser plugin apis, but I failed to find one.

The expected use case is using the parser of swc as a replacement for the babel parser.
To achieve max performance, I think the parser plugin should return code after applying compatibility transforms of swc.

Notes:

- swc-to-babel ast converter is WIP at https://github.com/swc-project/swc/pull/1465

[swc]: https://swc.rs

**Comment by nicolo-ribaudo** [comment]
You can use the `parserOverride` plugin property. Example: https://lihautan.com/creating-custom-javascript-syntax-with-babel/#our-babel-plugin

**Reply by kdy1** [reply]
Thanks!
